Updated homework model and added homework controller and homework view

bradweee opened the pull request:
**Changes**

- Updated the homework model.
- Added homeworks controller and homework view
- Updated lectures view to include link to add homework to lecture
- Updated homework form to input content, due date, and status
- Added better errors to gemfile, dont know if others will like it, but I do 😄 

**Open Issue**
The lecture id isn't getting passed through to the homework when clicking the "Create Homework" button. Error says it couldn't fine the lecture id although it should have been passed through with set_lecture (line 72 of homeworks_controller). Would love some help diagnosing the issue.

shaolinhoward commented on app/views/homeworks/index.html.erb:
```diff
@@ -0,0 +1,25 @@
+<p id="notice"><%= notice %></p>
+
+<h1>Homeworks</h1>
```
not plural :)

perhaps "Homework Assignments" 

will bust the test cases below

muffy replied:
Agreed.

bradweee replied:
Changed :)

muffy commented on app/controllers/homeworks_controller.rb:
```diff
@@ -0,0 +1,81 @@
+    # Only allow a list of trusted parameters through.
+    def homework_params
+      params.fetch(:homework, {})
+      # params.require(:homework).permit(:content, :status, :due_date)
```
You should definitely use the `permit` syntax here, it's important for security to avoid unexpected parameters.

muffy commented on app/views/homeworks/_form.html.erb:
```diff
@@ -0,0 +1,37 @@
+    <%= form.date_select :dute_date, :order => [:month, :day, :year] %>
+  </div>
+
+
```
Remove extra empty line here.

muffy commented on test/controllers/homeworks_controller_test.rb:
```diff
@@ -0,0 +1,48 @@
+require 'test_helper'
```
Good work on adding tests. Generally for tests we can assume that Rails will work as advertised, so try to make tests just for logic you have added, such as an input validation.

muffy commented on app/helpers/homeworks_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module HomeworksHelper
+end
```
Generally I would agree, but if you expect to need it in the VERY near future, leaving the boilerplate in place saves some trouble later. Totally agree on evaluating what you actually need right now.

bradweee commented:
Resolving comments from previous commit:

- Removed extra carrot-u-admin-app filesRemoved .DS_Store & added file to .gitignore
- Switched to permit syntax for def_homework_params in homeworks_controller
- Removed extra line from homeworks/_form
- Edited <h1> in homeworks/index to "Homework Assignments"
- Removed homeworks helper
- DRY'd up homeworks_controller update, create (includes an install of responders gem)

viktor-evdokimov commented on .gitignore:
```diff
@@ -40,3 +40,6 @@ yarn-debug.log*
 .idea
+
+#Ignore .DS_Store
+/app/.DS_Store
```
make it just .DS_Store and remove one from the root too =)
